packages/shared/src/components/modals/ArticlePostModal.tsx

```diff
@@ -1,21 +1,34 @@
 import React, { ReactElement, useContext } from 'react';
 import { Modal, ModalProps, modalSizeToClassName } from './common/Modal';
-import {
-  ONBOARDING_OFFSET,
-  PassedPostNavigationProps,
-  PostContent,
-} from '../post/PostContent';
+import { ONBOARDING_OFFSET, PostContent } from '../post/PostContent';
 import { Origin } from '../../lib/analytics';
 import usePostNavigationPosition from '../../hooks/usePostNavigationPosition';
 import BasePostModal from './BasePostModal';
 import OnboardingContext from '../../contexts/OnboardingContext';
 import { Post, PostType } from '../../graphql/posts';
+import CollectionPostContent from '../post/collection/CollectionPostContent';
+import { PassedPostNavigationProps } from '../post/common';
 
 interface ArticlePostModalProps extends ModalProps, PassedPostNavigationProps {
   id: string;
   post: Post;
 }
 
+type SimilarPostTypes = Exclude<
```

Comment:
Decided to use the ArticlePostModal as the content seemed to be similar and to not replicate more code. Do change if more changes are required.